src/main/java/in/pune/royforge/eledgerapi/data/dao/TransactionDAOImpl.java

```diff
@@ -12,11 +12,30 @@
 
 @Repository
 public class TransactionDAOImpl implements ITransactionDAO {
+
 	@Autowired
 	ITransactionLogRepository transactionLogRepository;
-	
-	
-	//Method is used to display the list of transaction logs
+
+	/*
+	 * Method to return a list List contains the details of transaction done between
+	 * a particular lender ID and borrower ID Lender Id and Borrower ID is passed as
+	 * argument to this method
+	 * 
+	 */
+	@Override
+	public List<Transaction> getTransactionsUsingLenderIdAndBorrowerId(String lenderId, String borrowerId) {
+		List<Transaction> transactions = new ArrayList<>();
+		Iterable<TransactionEntity> transactionsList = transactionLogRepository.transactionsList(lenderId, borrowerId);
```

Comment:
check isEmpty now like -----> 

List<Transaction> transactions = new ArrayList<>();
		List<TransactionEntity> transactionEntities = transactionLogRepository.transactionsList(lenderId, borrowerId);
		if (transactionEntities.isEmpty()) {
			for (TransactionEntity transaction1 : transactionEntities) {
				Transaction transactionInfo = new Transaction();
				setTransactionData(transaction1, transactionInfo);
				transactions.add(transactionInfo);
			}
		}
		return transactions;